oauth2-server/src/main/java/com/clouway/oauth2/ClientAuthorizationActivity.java

```diff
@@ -46,7 +46,7 @@ public Response execute(String identityId, Request request) {
     Optional<Client> possibleClientResponse = clientFinder.findClient(clientId);
 
     if (!possibleClientResponse.isPresent()) {
-      return OAuthError.unauthorizedClient();
+      return OAuthError.unauthorizedClient("A client with id:" + clientId + " was either not found or is not authorized.");
```

Comment:
We should be careful with these messages as someone could use them for a brute-force. 

```java
OAuthError.unauthorizedClient(String.format("Unknown client '%s'", clientId))
```

Reply:
Changed. We may omit the clientId, but that is an information we are already exposing on login.